server/src/main/java/org/apache/uniffle/server/storage/MultiStorageManager.java

```diff
@@ -69,35 +71,44 @@ public void updateWriteMetrics(ShuffleDataFlushEvent event, long writeTime) {
   }
 
   @Override
-  public boolean write(Storage storage, ShuffleWriteHandler handler, ShuffleDataFlushEvent event) {
+  public boolean write(Storage storage, ShuffleWriteHandler handler, ShuffleDataFlushEvent event, 
+                       CreateShuffleWriteHandlerRequest request) {
     StorageManager storageManager = selectStorageManager(event);
-    if (storageManager == coldStorageManager && event.getRetryTimes() > fallBackTimes) {
```

Comment:
In the original fallback strategy, blocks will not be written to `coldStorageManager` after `retryTimes > fallBackTimes` again. How about If `warmStorageManager` can't be written too.

Reply:
Directly fast fail ?

Reply:
Is there a possibility that the disk is temporarily full and the network is temporarily unavailable?

Reply:
I think the original retryTimes fallback is for the big data block directly flushed to HDFS. If we preserve this mechanism, we also should preserve this fallback strategy. Is it involved in this strategy?

Reply:
I think the original mechanism is for when hdfs can't be writed. And i think is it involved in this strategy. In this pr, if you want to write to hdfs as much as possible when hdfs temporarily can't be write, you can set `retryTimes` to a larger value, and it will be same as original mechanism when `retryTimes` is equal to `retryMax`

Reply:
> I think the original mechanism is for when hdfs can't be writed. And i think it is involved in this strategy. In this pr, if you want to write to hdfs as much as possible when hdfs temporarily can't be write, you can set `fallBackTimes` to a larger value, and it will be same as original mechanism when `fallBackTimes` is equal to `retryMax`

I was wrong. In the original mechanism, if one event fail to write to hdfs, it will not try it again, i think it is not a good mechanism. should it preserved?

Reply:
If shuffle server have many disk errors, they will fallback many small io write request to HDFS, HDFS may be influenced. So not all people want to fallback from local file to HDFS.  Maybe we need an configuration option to make users to select his best strategy.

Reply:
How about add two more strategies? 
1. Just allow  fallback from local file to HDFS;
2. Just allow  fallback from HDFS to local file.

Reply:
OK